README.md

```diff
@@ -1,298 +1,433 @@
-**callback** function. Each invocation of **callback** is called with three
-arguments: (element, index, collection). If **collection** is a JavaScript
-object, **callback**'s arguments will be (value, key, collection). **Returns
-the original collection for chaining.**
+Example function calls:
 
 ```javascript
-fi.each([1, 2, 3], alert);
+myEach([1, 2, 3], alert);
 => alerts each number in turn and returns the original collection
-fi.each({one: 1, two: 2, three: 3}, alert);
+
+myEach({one: 1, two: 2, three: 3}, alert);
 => alerts each number value in turn and returns the original collection
 ```
 
-**fi.map**
+### myMap
+
+`myMap(collection, callback)`
+
+Argument(s):
+
+- a collection (either an object or an array)
+- a callback function
+
+Return value:
+
+- A new array
 
-`fi.map(collection, callback)`
+Behavior:
 
-Produces a new array of values by mapping each value in **collection** through
-a transformation function (**callback**). The callback is passed three
-arguments: the value, then the index (or key) of the iteration, and finally a
-reference to the entire collection. **Returns a new collection for chaining
-without modifying the original.**
+Produces a new array of values by mapping each value in `collection` through a
+transformation function, `callback`. Returns the new array for chaining without
+modifying the original.
+
+Example function calls:
 
 ```javascript
-fi.map([1, 2, 3], function(num){ return num * 3; });
+myMap([1, 2, 3], function(num){ return num * 3; });
 => [3, 6, 9]
-fi.map({one: 1, two: 2, three: 3}, function(num, key){ return num * 3; });
+
+myMap({one: 1, two: 2, three: 3}, function(num, key){ return num * 3; });
 => [3, 6, 9]
 ```
 
-**fi.reduce**
+### myReduce
+
+`myReduce(collection, callback, acc)`
+
+Argument(s):
+
+- a collection (either an object or an array)
+- a callback function
+- a starting value for the accumulator
+
+Return value:
+
+- A single value
 
-`fi.reduce(collection, callback, acc)`
+Behavior:
 
-Reduce boils down a **collection** of values into a single value. **Acc**
-(short for accumulator) starts as the initial state of the reduction, and with
-each successive step it should be accumulate the return value of **callback**.
-The callback is passed three arguments: the acc, the current value in our
-iteration (the element in the array), and finally a reference to the entire
+Reduce iterates through a `collection` of values and boils it down into a single
+value. `acc` (short for accumulator) starts at the value that's passed in as an
+argument, and with each successive step is updated to the return value of
+`callback`. If a start value is not passed to `myReduce`, the first value in
+the collection should be used as the start value.
+
+The `callback` is passed three arguments: the current value of `acc`, the
+current element/value in our iteration, and a reference to the entire
 collection.
 
+**Hint:** For the case when a start value is not passed in as an argument, think
```

Comment:
Add: "when a start value **for the accumulator** is not passed in"

Reply:
Done